CONTRIBUTING.md

```diff
@@ -0,0 +1,373 @@
+
+[Leave the code better than you found it...](https://martinfowler.com/bliki/OpportunisticRefactoring.html)
+in the context of what you're addressing.
+
+Boy Scouts have a rule that says: "Always leave the campground cleaner than you found it",
+which does not mean, "leave the whole forest cleaner than you found it". 
+
+#### Keep the pull request focused
+
+A pull request should solve only one thing. This makes it easier to review and share, reduce
+risk and criticality and improve understanding of the changes. If you find yourself in
+a position of doing more than needed, even if this is something trivial but not directly
+related to your current work, come back later! Open a new issue and come back later to address
+that problem. Follow-ups are welcome!
+
+#### Open follow-ups
+
+We know it. It is hard just to ignore that something else can be improved. Don't do it.
+
+When you foresee that addressing a problem will take more effort than documenting it,
+choose the latter and open a follow-up. When assessing the effort, think twice:
+1. Once for you to make the change.
+2. Another for the reviewer to switch context and review the changes.
+
+#### Review in layers
+
+When reviewing, highlighting all the things that you consider can be improved may be
+exhausting for the contributor. Instead of noticing everything at the same time, try to layer
+your review.
+
+1. Start by testing the branch and indicating errors in the implementation.
+2. Continue asking for clarifications about what you don't understand.
+3. Then, address semantic and architectural problems including type annotations that will result in the hardest-to-remove technical debt in the future.
+4. Now, you can focus on readability: typos, variable names, team idioms, style violations not caught by the automated checks.
+5. Finally, go for other less critical code smells.
+
+#### Make branch names meaningful
+
+Include your initials, the number of the issue or both. That helps to identify who started
+the branch and what issue is being addressed:
+
+```
+sp-issue-1234-implement-new-header
+```
+
+#### Sanitize your branch history
+
+Rebase on your own history to reorder, fix and squash your history. Remember that each commit
+should represent a meaningful atomic change. For instance, if you commit a new functionality
+just to realize it is broken because you misspelt a variable, add another commit for fixing it,
+then rebase and combine both commits together.
+
+## Code style
+
+### Type annotations
+
+### Writing components
+
+## Project management
+
+In Qiskit, we take open-source seriously and want to be open and transparent to
+our community. This is why we keep most of our management public, with the help
+of [projects](https://github.com/Qiskit/qiskit.org/projects) for sprint planning and
+[issues](https://github.com/Qiskit/qiskit.org/issues) for tracking what needs to
+be done.
+
+We use a customized version of [scrum](https://www.atlassian.com/agile/scrum) with
```

Comment:
I think the "how we work during the sprint" should be in a separate document available only for the team members. IMO, this is something a external contributor doesn't need to know to contribute